mava/utils/networks_utils.py

```diff
@@ -0,0 +1,155 @@
+            layer_norm: apply layer normalisation hidden MLP layers.
+            name: Optional name for this module.
+        Raises:
+            ValueError: If ``with_bias`` is ``False`` and ``b_init`` is not ``None``.
+        """
+        if not with_bias and b_init is not None:
+            raise ValueError("When with_bias=False b_init must not be set.")
+
+        super().__init__(name=name)
+        self.with_bias = with_bias
+        self.w_init = w_init
+        self.b_init = b_init
+        self.activation = activation
+        self.activate_final = activate_final
+        self.layer_norm = layer_norm
+        layers = []
+        norms = []
+        output_sizes = tuple(output_sizes)
+        for index, output_size in enumerate(output_sizes):
+            layers.append(
+                hk.Linear(
+                    output_size=output_size,
+                    w_init=w_init,
+                    b_init=b_init,
+                    with_bias=with_bias,
+                    name="linear_%d" % index,
+                )
+            )
+            norms.append(
+                hk.LayerNorm(
+                    axis=-1,
+                    create_scale=True,
+                    create_offset=True,
+                    param_axis=-1,
+                    use_fast_variance=False,
+                    name="norm_%d" % index,
+                )
+            )
+
+        self.layers = tuple(layers)
+        self.norms = tuple(norms)
+        self.output_size = output_sizes[-1] if output_sizes else None
+
+    def __call__(
+        self,
+        inputs: jnp.ndarray,
+        dropout_rate: Optional[float] = None,
+        rng: Any = None,
+    ) -> jnp.ndarray:
+        """Connects the module to some inputs.
+        Args:
+            inputs: A Tensor of shape ``[batch_size, input_size]``.
+            dropout_rate: Optional dropout rate.
+            rng: Optional RNG key. Require when using dropout.
+        Returns:
+            The output of the model of size ``[batch_size, output_size]``.
+        """
+        if dropout_rate is not None and rng is None:
+            raise ValueError("When using dropout an rng key must be passed.")
+        elif dropout_rate is None and rng is not None:
+            raise ValueError("RNG should only be passed when using dropout.")
+
+        rng = hk.PRNGSequence(rng) if rng is not None else None
+        num_layers = len(self.layers)
+
+        out = inputs
+        for i, layer in enumerate(self.layers):
+            out = layer(out)
+            if i < (num_layers - 1) or self.activate_final:
+                # Only perform dropout if we are activating the output.
+                if dropout_rate is not None:
+                    out = hk.dropout(next(rng), dropout_rate, out)  # type: ignore
+                out = self.activation(out)
+            if self.layer_norm:
+                out = self.norms[i](out)
+
+        return out
+
+    def reverse(
```

Comment:
Is this function used internally when running our examples? Or is it required?

Reply:
No we don't use it. I have removed it. Also it was the one reducing the test coverage.